File: MATLAB/Modele_4/verifierSpTr.m
function [isInWorkspace] = verifierSpTr(x,y)
    parametres;
    funct1=sqrt((x+L)^2+y^2);
    funct2=sqrt((x-L)^2+y^2);
    if funct1<r_eq_min || funct1>r_eq_max
        isInWorkspace = 0;
        warning("Some points are not on the workspace")
    elseif funct2<r_eq_min || funct2>r_eq_max
        isInWorkspace = 0;
        warning("Some points are not on the workspace")
    else
        isInWorkspace = 1;
    end
end

File: EQRIGIDE/verifierSpTr.py
import numpy as np
import warnings
import parametres as p

def verifierSpTr(x, y):
    funct1 = np.sqrt((x + p.L)**2 + y**2)
    funct2 = np.sqrt((x - p.L)**2 + y**2)
    
    if funct1 < p.r_eq_min or funct1 > p.r_eq_max:
        warnings.warn("Certains points ne se trouvent pas dans l'espace de travail.")
        return 0
    elif funct2 < p.r_eq_min or funct2 > p.r_eq_max:
        warnings.warn("Certains points ne se trouvent pas dans l'espace de travail.")
        return 0
    else:
        return 1